tests/vcs/test_vcs.py

```diff
@@ -285,6 +302,17 @@ def test_normalize_url(url, normalized):
                 name="blah",
             ),
         ),
+        (
+            "git+https://fafb334-cb038533f851c23d0b63254223Abf72ce4f02987e7064b0c95566699a:x-oauth-basic@hostname/project/blah.git",
+            ParsedUrl(
+                protocol="https",
+                resource="hostname",
+                pathname="/project/blah.git",
+                user="fafb334-cb038533f851c23d0b63254223Abf72ce4f02987e7064b0c95566699a",
+                password="x-oauth-basic",
```

Comment:
This is not entirely correct since the `user` identified here is actually the personal access token, and the `password` is the auth protocol. However, it was the easiest thing to do without rewriting the regex matching and url parsing logic.